src/worker.js

```diff
@@ -156,7 +156,78 @@ self.freeTrack = () => {
  * @param {!string} url the URL of the subtitle file.
  */
 self.setTrackByUrl = ({ url }) => {
-  self.setTrack({ content: read_(url) })
+  // simulate throttling due to network or resource pressure
+  // let delay = 0;
+  let fontnameIndex;
+
+  const process = line => {
+    preprocess: if (availableFonts) {
+      // Ignore empty lines and comments
+      if (!line || line.startsWith(';')) {
+        break preprocess;
+      }
+
+      // Reset format with new sections
+      if (line.startsWith('[')) {
+        fontnameIndex = undefined;
+        break preprocess;
+      }
+
+      const verb = line.split(':')[0];
+
+      if (verb === 'Format') {
+        fontnameIndex = line.split(',').findIndex(s => s.trim() === 'Fontname');
+      } else if (verb === 'Style') {
+        let font = line.split(',')[fontnameIndex].trim();
+
+        if (font) {
+          findAvailableFonts(font);
+        }
+      }
+
+      const regex = /\\fn([^\\}]*?)[\\}]/g;
+      let matches;
+      while ((matches = regex.exec(line)) !== null) {
+        findAvailableFonts(matches[1])
+      }
+    }
+
+    // delay += 200 + Math.random() * 500;
+    // setTimeout(() => {
+    if (dropAllBlur) line = dropBlur(line)
+    jassubObj.processLine(line);
+    // }, delay);
+  };
+
+  _fetch(url).then(response => {
+    let partialLine = '';
+
+    return response.body
+      .pipeThrough(new TextDecoderStream())
+      .pipeTo(new WritableStream({
```

Comment:
I'm debating if we should use this, or async iterators, also the issue becomes that this isn't supported on legacy devices like webos

Reply:
I added a check to fallback if they don't exist. Using async iterators makes the code a bit larger and worse imo

Reply:
async iterators would require a polyfill, so instead lets do this with a while loop
```js
const res = await fetch(location)
const reader = res.body.pipeThrough(new TextDecoderStream()).getReader()

jassubObj.newEmptyTrack();
subtitleColorSpace = libassYCbCrMap[jassubObj.trackColorSpace];
postMessage({ target: 'verifyColorSpace', subtitleColorSpace });

let partialLine = ''
while (true) {
  const {value, done} = await reader.read()
  if (done) break
  const lines = value.split(/\r\n|[\r\n]/g)
  
  partialLine = lines[lines.length-1]
  
  for (const line of lines.slice(0, -1)) {
     process(line);
  }
}
process(partialLine)
```

Reply:
there's still a bug here that if one changes track while one is being processed the changed track will get data from the old one, that needs to abort